recipes/opencv/2.x/conanfile.py

```diff
@@ -0,0 +1,263 @@
+
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["BUILD_WITH_STATIC_CRT"] = "MT" in str(self.settings.compiler.runtime)
+        self._cmake.configure(build_folder=self._build_subfolder)
+
+        return self._cmake
+
+    def build(self):
+        self._patch_opencv()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "staticlib"))
+        tools.remove_files_by_mask(self.package_folder, "*.cmake")
+
+    def package_info(self):
+        version = self.version.split(".")[:-1]  # last version number is not used
+        version = "".join(version) if self.settings.os == "Windows" else ""
+        debug = "d" if self.settings.build_type == "Debug" and self.settings.compiler == "Visual Studio" else ""
+
+        def get_lib_name(module):
+            return "opencv_%s%s%s" % (module, version, debug)
+
+        def add_components(components):
+            # TODO: OpenCV doesn't use cmake target namespace
+            for component in components:
+                conan_component = component["target"]
+                cmake_target = component["target"]
+                lib_name = get_lib_name(component["lib"])
+                requires = component["requires"]
+                self.cpp_info.components[conan_component].names["cmake_find_package"] = cmake_target
+                self.cpp_info.components[conan_component].names["cmake_find_package_multi"] = cmake_target
+                self.cpp_info.components[conan_component].libs = [lib_name]
+                self.cpp_info.components[conan_component].requires = requires
+                if self.settings.os == "Linux":
+                    self.cpp_info.components[conan_component].system_libs = ["dl", "m", "pthread", "rt"]
+
+                # CMake components names
+                conan_component_alias = conan_component + "_alias"
+                cmake_component = component["lib"]
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package"] = cmake_component
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package_multi"] = cmake_component
+                self.cpp_info.components[conan_component_alias].requires = [conan_component]
+                self.cpp_info.components[conan_component_alias].includedirs = []
+                self.cpp_info.components[conan_component_alias].libdirs = []
+                self.cpp_info.components[conan_component_alias].resdirs = []
+                self.cpp_info.components[conan_component_alias].bindirs = []
+                self.cpp_info.components[conan_component_alias].frameworkdirs = []
```

Comment:
Just to confirm since I never see these... are they required?

Reply:
Those attributes are not empty be default. Required? No really, I would say more optimal maybe.